doc/content.adoc

```diff
@@ -311,26 +311,38 @@ user defined types to jooq/jdbc compatible types, and other for backwards conver
 .Let see an example for adapt clojure persistent map interface to postgresql json file.
 [source, clojure]
 ----
-(require '[suricatta.proto :as proto])
-(require '[cheshire.core :as json])
+(require '[suricatta.proto :as proto]
+         '[suricatta.impl :as impl]
+         '[cheshire.core :as json])
+
 (import 'org.postgresql.util.PGobject)
 
 (extend-protocol proto/IParamType
   clojure.lang.IPersistentMap
-  (-render [self]
-    (str "'" (json/generate-string self) "'::json"))
-  (-bind [self stmt index]
-    (let [obj (doto (PGobject.)
-                (.setType "json")
-                (.setValue (json/generate-string self)))]
-      (.setObject stmt index obj))))
+  (-render [self ctx]
+    (if (impl/inline? ctx)
+      (str "'" (json/encode self) "'::json")
+      "?::json"))
+
+  (-bind [self ctx]
+    (when-not (impl/inline? ctx)
+      (let [stmt (.statement ^BindContext ctx)]
```

Comment:
I understand your point of the view, but suricatta does not intend to hide jOOQ. It just intends to offer a good clojure friendly api with some "opinion". In normal use yous should not care about jooq but extending it with custom types is considered "advanced stuff" and should be done once, with the power to access to the internals. 

A good solution maybe for it would be provide a protocol for access the essential properties from the context making the "usable" api explicit but not limited to it... (because you will continue be able access to the object properties).